Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -151,12 +152,125 @@ class UpstreamIntegrationTests: XCTestCase {
+            "arrayOfObjectsPass": [
+              {
+                "object1": "value1"
+              },
+              {
+                "object2": "value2"
+              }
+            ],
+            "arrayOfObjectsFail": [
+              {
+                "object1": "value1"
+              },
+              {
+                "object2": "value2"
+              }
+            ],
+            "dictionaryCompare": {
+              "nested1": "value1"
+            },
+            "dictionaryNestedCompare": {
+              "nested1": {
+                "nested2": {
+                  "nested3": {
+                    "nested4": {
+                      "nested1": "value1"
+                    }
+                  }
+                }
+              }
+            },
+            "trulyNested": [
+              {
+                "nest1": [
+                  {
+                    "nest2": {
+                      "nest3": [
+                        {
+                          "nest4": "value1"
+                        }
+                      ]
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+        """#
+        
+        let multilineInput = #"""
+          {
+            "integerCompare": 0,
+            "decimalCompare": 4.123,
+            "stringCompare": "def",
+            "boolCompare": false,
+            "nullCompare": "not null",
+            "arraySizeCompare": [1,2],
+            "arrayValueCompare": [0],
+            "arrayOfObjectsPass": [
+              {
+                "object1": "value1"
+              },
+              {
+                "object2": "value2"
+              }
+            ],
+            "arrayOfObjectsFail": [
+              {
+                "object1": "value1"
+              },
+              {
+                "object2": "value3"
+              }
+            ],
+            "dictionaryCompare": {
+              "nested1different": "value1"
+            },
+            "dictionaryNestedCompare": {
+              "nested1": {
+                "nested2": {
+                  "nested3": {
+                    "nested4": {
+                      "nested1": "value2"
+                    }
+                  }
+                }
+              }
+            },
+            "trulyNested": [
+              {
+                "nest1": [
+                  {
+                    "nest2": {
+                      "nest3": [
+                        [
+                          "nest4"
+                        ]
+                      ]
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+        """#
         
-        XCTAssertEqual("success", "not success")
-        XCTFail("This is an example failure message: \(experienceEvent)")
+        // MARK: AnyCodable testing
+        let jsonValidation = try? JSONDecoder().decode(AnyCodable.self, from: multilineValidation.data(using: .utf8)!)
```

Comment:
For JSON comparison you can decode straight to [String, AnyCodable].self and have direct access to the keys if needed